### ISSUE TITLE
Change name from "Free Code Camp" to freeCodeCamp everywhere

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/freeCodeCamp#12472]
We're changing our community's written name from "Free Code Camp" to "freeCodeCamp". 

In case your curious, here are some of the reasons our core team agreed that this way of writing it out is better:

- most people already type freecodecamp when they search for us on Google (via Google Trends), and 60% of them self-report that they type it without spaces (via Twitter poll)
![](https://www.evernote.com/l/Ar879NtdSjZGy5KcjkUT_zwUDD_ZBcX0R8YB/image.png)
![](https://www.evernote.com/l/Ar_Q2orA4I5PQ5sJTRIxlAVdGEujsd5nJR0B/image.png)

- freecodecamp is easier and faster to type, and visually more 

- SEO practitioners have told me that just using one name (not both) will improve our SEO (though Google is basically a black box, so there's no way to know whether they're right)

- with the optional lower-camel-casing, freeCodeCamp looks visually more interesting than Free Code Camp without singificantly affecting readability

- our logo, twitter handle, facebook page, youtube channel, medium publication - you name it - already use freecodecamp without spaces, so this is a step toward uniformity

Here are the places where we will need to update this:
- [x] Our forum
- [x] Gitter descriptions
- [x] npm packages
- [x] Twitter
- [x] Facebook
- [x] Medium publication
- [x] GitHub README.md and other .md files
- [x] References within Free Code Camp's codebase itself, including meta tags and challenge copy
- [x] YouTube channel name and information
- [ ] YouTube video descriptions (these were slated to be updated anyway)
- [x] LinkedIn (note that LinkedIn is transitioning their universities pages platform so we weren't able to change everything yet)
- [x] Teespring
- [x] codepen
- [x] reddit

And anywhere else anyone notices it. 

If you want to tackle a specific platform listed above, reply and call "dibs." And if you don't have permission to update it, reply to this issue with links to specific instances and someone will probably have permissions to go and update it.

Thanks for your help with this transition, everyone!

**Comment by Manish-Giri** [comment]
I call dibs on 'GitHub README.md and other .md files'!

+1 for the camelCase!

**Comment by erictleung** [comment]
The reddit page also needs to be changed https://www.reddit.com/r/FreeCodeCamp/. The URL, the page title, and the description need changing.

**Comment by raisedadead** [comment]
Updated, Codepen.io : Username / Display Name / Challenge Pen Titles / Icons

**Comment by no-stack-dub-sack** [comment]
Surprisingly, I found only 289 matches for "free code camp" (non-case-sensitive) across 40495 files in all of `FreeCodeCamp/`

**Comment by no-stack-dub-sack** [comment]
@QuincyLarson @raisedadead @erictleung, I'd like to start replacing some of these in the codebase in places where I know it won't be an issue, for instance, I just made 56 replacements in `FreeCodeCamp/server/resources/testimonials.json` - at what point should I make a PR for this?

**Comment by HKuz** [comment]
The new challenges in the curriculum expansion that reference it use "Free Code Camp" in their copy. I think I based it off the front-end sections that use the "Free Code Camp Cat Photo API". Happy to help get those when The Merge is ready.

**Comment by raisedadead** [comment]
@no-stack-dub-sack I recommend, that you make a pull request per file/block as in one for all front end projects, one for all basic javascript and so on.

This will help a lot in rebasing the current open developments. (Also, it would be a life saver in QA, scrolling is a tough lol... 😅 )

**Comment by raisedadead** [comment]
@HKuz I think for the curriculum expansion we can have a separate PR that updates that branch, we need not wait for staging. It's ideal because the changes from curriculum branch would go on top of staging, so ideally nothing would be missed.

Let me knw, I can make a PR to help.

**Comment by Manish-Giri** [comment]
@raisedadead @erictleung Are there any other platform(s) that I can take up from the list above, without running into access issues?

**Comment by raisedadead** [comment]
@Manish-Giri how about forum?

**Comment by Manish-Giri** [comment]
@raisedadead Thanks! I'll go through the forum.

**Comment by no-stack-dub-sack** [comment]
@raisedadead So I was talking to @QuincyLarson about this last night actually, and he mentioned that perhaps I shouldn't touch the seed files since they will all be replaced when the curriculum expansion goes live anyway - did I understand that correctly @QuincyLarson?

So far I've done the following directories:

`server/resources/testimonials .json`
`server/views/sponsors/sponsors.jade`
`server/boot/user.js`
`server/boot/a-extender.js`
`server/boot/randomAPIs.js`
`server/server.js`

and I did this before I talked to Quincy:

`seed/challenges/04-video-challenges/jslingo.json`

So I guess I will finish the `server` directory and then make a PR, then do everything else directory by directory, except in the seed files, and file a PR for each.

Make sense @raisedadead?

**Comment by raisedadead** [comment]
> I shouldn't touch the seed files since they will all be replaced when the curriculum expansion goes live anyway

Well this has a potential of missing out and NOT touching some lines or instances or occurrence of Free Code Camp or FreeCodeCamp 

IMHO, we should have two activities of updating both branches on a chore activity. But yeah that will be identified easily as well, later on... no worries.

Let's make the changes for seed in the curriculum branch and audit everything once it's merged back to staging.

> then do everything else directory by directory, except in the seed files, and file a PR for each

Perfect. Sounds good to me. Just check for open PRs, if any PR is having changes (files) that will be affected with your PR, refer to it in the PR comments, so that the PR owners can rebase and resolve conflicts.

> Also, since the domain part of emails addresses is case insensitive, should I be replacing "freecodecamp" with "freeCodeCamp" in any instance of an email address that a user might see, or is that overkill?

No. DO NOT do that. It may not be an issue in browsers but there could be email clients that can break... also its an overkill.

Thanks.

**Comment by robbawebba** [comment]
@QuincyLarson @raisedadead @erictleung I noticed that the freeCodeCamp listing on [up-for-grabs.net](http://up-for-grabs.net) is using the old style for freeCodeCamp. Do you guys mind if I change it to the new style? I'll have to change [this file](https://github.com/up-for-grabs/up-for-grabs.net/blob/gh-pages/_data/projects/Free_Code_Camp.yml) and submit the PR to the [up-for-grabs.net repo](https://github.com/up-for-grabs/up-for-grabs.net)

**Comment by Manish-Giri** [comment]
@raisedadead As I was working through renaming in the `.md` files, I came across [this file](https://github.com/freeCodeCamp/freeCodeCamp/blob/782f489576db3fa4ec63ed842d3b256a759d3f3a/LICENCE-Free-Code-Camp-Curriculum.md). The file name is `LICENCE-Free-Code-Camp-Curriculum.md`, and as the file will be renamed according to the new style, should I change `LICENCE` to `LICENSE`, while I'm at it? I noticed, in some other places, `License` is being used, like [this file](https://github.com/freeCodeCamp/freeCodeCamp/blob/782f489576db3fa4ec63ed842d3b256a759d3f3a/README.md#license), so wanted to check with you.

**Comment by erictleung** [comment]
@robbawebba that sounds reasonable!

@Manish-Giri in my opinion, I was thinking the license file for the curriculum could be changed to `LICENSE-freeCodeCamp-Curriculum.md`, which uses `freeCodeCamp` and `LICENSE` over `LICENCE`. My two cents, but I'd wait on @QuincyLarson.

**Comment by erictleung** [comment]
FYI, the forum [About](https://forum.freecodecamp.com/about) page need to be updated.

**Comment by raisedadead** [comment]
@Manish-Giri Please proceed as Eric suggested. I must have messed up the spelling in the first place.

@robbawebba yes, thanks for that!

@m-henderson can you help with about page for the forum.

**Comment by Manish-Giri** [comment]
@erictleung @raisedadead Thanks guys! I'll wait on @QuincyLarson's final word for the change.

**Comment by m-henderson** [comment]
@raisedadead of course, bud. I updated the forum about page from "The Free Code Camp Forum" to "The freeCodeCamp Forum"

**Comment by QuincyLarson** [comment]
@erictleung I updated our subreddit. Reddit won't let you change the capitalization of the URL but I fixed everything else.

**Comment by no-stack-dub-sack** [comment]
@raisedadead @QuincyLarson with the PR I just made, all that is really left for the codebase is the `.md` files which I know @Manish-Giri is on, and the `seed` files - which I guess we are not doing for now, right? Or should I go ahead and make a PR against the curriculum branch?

**Comment by raisedadead** [comment]
@no-stack-dub-sack yup, sounds good.

**Comment by raisedadead** [comment]
@Manish-Giri whenever you are ready, we need that PR, :wink:

**Comment by no-stack-dub-sack** [comment]
@QuincyLarson @raisedadead 

Two issues/comments/questions:
1. I am going to leave any `alt` attributes I come across as "free code camp" since we want to ensure screen readers can pronounce this.
2. Challenge [caesars cipher](https://www.freecodecamp.com/challenges/caesars-cipher) relies on a spelling of FREE CODE CAMP to pass the tests, so I am not changing this instance in our seed. I think it is probably not worth fixing, but wanted to mention it.

**Comment by systimotic** [comment]
Does anybody know whether https://github.com/freeCodeCamp/static is still in use? If so, it will also require some updates.

**Comment by systimotic** [comment]
Hello people! I checked the checklist to see what has been done, and what still has to be done.

**Our forum**
✔️ The title of the forum was updated
♻️ Wiki renaming is in progress

**Gitter descriptions**
✔️ I went through the [official chatrooms](https://forum.freecodecamp.com/t/free-code-camp-official-chat-rooms/19390) and changed these.

**npm packages**
❓ All freeCodeCamp packages are named either with fcc or lower case freecodecamp, as makes sense with the naming conventions for packages. The only thing I could find that may need updating is [fcc-express-background](https://www.npmjs.com/package/fcc-express-background). /cc @Em-Ant 

**Twitter**
and
**Medium publication**
✔️ Looks like this was done

**GitHub README.md and other .md files**
✔️ Done by Manish

**Facebook**
♻️❓ Looks like the description was updated, but the title hasn't been updated yet. From what I can tell, you have to do this through a request, so there's a chance this is already in progress, but I'm not sure.

**References within Free Code Camp's [sic 😜] codebase itself, including meta tags and challenge copy**
♻️ In progress, looks like the challenge part is waiting for https://github.com/freeCodeCamp/freeCodeCamp/pull/12570 and everything else is done.

### Todo
YouTube channel name and information
YouTube video descriptions (these were slated to be updated anyway)
LinkedIn
Teespring / Shop


That's great progress! Most of the name change went through in one week, nice work everyone!

@QuincyLarson I'm not sure of the right places and people to take these last bits on. I think most of them are up to you.

**Comment by raisedadead** [comment]
@QuincyLarson @systimotic PR https://github.com/freeCodeCamp/shop/pull/1 up for the shop.

**Comment by Dazler** [comment]
Hi everyone! I have made some commits into my branch and trying to merge with master!! But it is asking for a reason ,,, i i have  no idea how to continue ahead!! I have attached the screenshot.Please help.
![ice_screenshot_20170121-144354](https://cloud.githubusercontent.com/assets/21132868/22173425/48d0cb88-dfe8-11e6-9ea2-ec5713916f8b.png)

**Comment by Dazler** [comment]
I was making changes in seed/challenges files which were using old style of "freeCodeCamp".

**Comment by raisedadead** [comment]
@Dazler  reachout to the [Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors).

**Comment by no-stack-dub-sack** [comment]
I can go ahead and finish the codebase now that we've made the big curriculum merge into staging... I will work on a PR to finish this off in the next few days. Which is just seed files at this point for the codebase.

**Comment by no-stack-dub-sack** [comment]
@raisedadead I've been holding off on this task for a while, since there are SOOOOO many changes going on in the seed files right now... not sure when a good time to do this would be. I'm afraid it would cause merge conflicts with open PRs, etc.

**Comment by raisedadead** [comment]
@no-stack-dub-sack please make a pull request when ready, we will take care of merging and rebasing PRs

**Comment by hisener** [comment]
Could someone change the description of the [about repo](https://github.com/freeCodeCamp/about)?

freeCodeCamp/about#3

**Comment by aj-willi** [comment]
i would like to contribute, where can i start? 💃

**Comment by aj-willi** [comment]
i have noticed the lessons also will need to reworded with the new freeCodeCamp phrase, how can I help with this?

**Comment by no-stack-dub-sack** [comment]
@QuincyLarson @raisedadead Ah! I got one more place - Twitch! How could we forget!? 😄  Not sure who is in charge of this, but I just want to know for sure whether or not we can do it, because if I change to instructions for the twitch challenge to `freeCodeCamp` for "users who are usually online" not sure if the API will be sensitive to that case change or not...

**Comment by no-stack-dub-sack** [comment]
A beautiful site!

![image](https://cloud.githubusercontent.com/assets/18563015/22533644/50dccb50-e8bd-11e6-899c-03dcce81bd98.png)

**Comment by QuincyLarson** [comment]
@no-stack-dub-sack thanks! I'll take care of Twitch :)

**Comment by chrisscotthall** [comment]
How can I help out?

**Comment by QuincyLarson** [comment]
All right everyone, the only thing left on this list is the YouTube descriptions. I'm going to create a separate issue for that. Thanks everyone!
